content/blog/nuttx-adding-porting-an-app/index.md

```diff
@@ -0,0 +1,473 @@
+
+{{< highlight shell >}}
+menuconfig CRYPTO_MBEDTLS
+	bool "Mbed TLS Cryptography Library"
+	default n
+	---help---
+		Enable support for Mbed TLS.
+
+if CRYPTO_MBEDTLS
+
+config MBEDTLS_VERSION
+	string "Mbed TLS Version"
+	default "3.4.0"
+
+config MBEDTLS_DEBUG_C
+	bool "This module provides debugging functions."
+	default DEBUG_FEATURES
+	---help---
+		This module provides debugging functions.
+
+config MBEDTLS_SSL_IN_CONTENT_LEN
+	int "Maximum length (in bytes) of incoming plaintext fragments."
+	default 16384
+	---help---
+		Maximum length (in bytes) of incoming plaintext fragments.
+
+config MBEDTLS_SSL_OUT_CONTENT_LEN
+	int "Maximum length (in bytes) of outgoing plaintext fragments."
+	default 16384
+	---help---
+		Maximum length (in bytes) of outgoing plaintext fragments.
+
+config MBEDTLS_SSL_SRV_C
+	bool "This module is required for SSL/TLS server support."
+	default y
+	---help---
+		This module is required for SSL/TLS server support.
+
+config MBEDTLS_PLATFORM_MEMORY
+	bool "Enable the memory allocation layer."
+	depends on MBEDTLS_PLATFORM_C
+	default n
+
+config MBEDTLS_ENTROPY_HARDWARE_ALT
+	bool "Uncomment this macro to let mbed TLS use your own implementation of a hardware entropy collector."
+	default n
+	depends on DEV_RANDOM
+	select MBEDTLS_NO_PLATFORM_ENTROPY
+
+config MBEDTLS_AES_ROM_TABLES
+	bool "Store the AES tables in ROM."
+	default n
+
+config MBEDTLS_NO_PLATFORM_ENTROPY
+	bool "Do not use built-in platform entropy functions."
+	default n
+
+config MBEDTLS_ECP_RESTARTABLE
+	bool "Enable the restartable ECC."
+	depends on MBEDTLS_ECP_C
+	default n
+
+config MBEDTLS_SELF_TEST
+	bool "Enable the checkup functions (*_self_test)."
+	default n
+{{< /highlight >}}
+
+#### Make.defs
+
+Finally, the last file in the folder is `Make.defs`. By comparing it with the one of the `Hello World!` example, the Make.defs file of the Mbed TLS additionally defines the header files (.h) of the library to be used by other NuttX applications by adding the content of the mbedtls/include folder to the `CFLAGS` and `CXXFLAGS` variables of the NuttX build system:
+
+{{< highlight make >}}
+ifneq ($(CONFIG_CRYPTO_MBEDTLS),)
+CONFIGURED_APPS += $(APPDIR)/crypto/mbedtls
+
+# Allows `<mbedtls/<>.h>` import.
+
+CFLAGS += ${INCDIR_PREFIX}$(APPDIR)/crypto/mbedtls/include
+CFLAGS += ${INCDIR_PREFIX}$(APPDIR)/crypto/mbedtls/mbedtls/include
+CFLAGS += ${DEFINE_PREFIX}MBEDTLS_CONFIG_FILE="<mbedtls/mbedtls_config.h>"
+
+CXXFLAGS += ${INCDIR_PREFIX}$(APPDIR)/crypto/mbedtls/include
+CXXFLAGS += ${INCDIR_PREFIX}$(APPDIR)/crypto/mbedtls/mbedtls/include
+CXXFLAGS += ${DEFINE_PREFIX}MBEDTLS_CONFIG_FILE="<mbedtls/mbedtls_config.h>"
+
+endif
+{{< /highlight >}}
+
+## Conclusion? (not yet!)
```

Comment:
Maybe you can use "to be continued"
## Conclusion (to be continued)

Reply:
Done!